### PR TITLE
dev, v3.0: add TiDB cluster auto failover file

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? 

add TiDB cluster auto failover file

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1464

### Which version does your change affect? 

dev, v3.0

### CODE REVIEW

File: dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+By default, `pdFailoverPeriod`, `tikvFailoverPeriod` and `tidbFailoverPeriod` are set to be 5 minutes, which is the waiting timeout after confirming the instance failure. After this time, TiDB Operator begins the automatic failover process.
+
+## Implementation principles
+
+There are three components in a TiDB cluster - PD, TiKV and TiDB, each of which has its own automatic failover policy. This section gives an in-depth introduction to the three policies.
+
+### PD automatic failover policy
+
+Assume that there are 3 nodes in a PD cluster, and if a PD node is down over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator makes this node go offline first, and creates a new PD node. At this time, there are 4 Pods existing in the cluster. After the failed PD node gets back to normal, TiDB Operator deletes the newly created node and the number of nodes gets back to 3.
```
Comment (dcalvin) [inline dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md]:
```suggestion
Assume that there are 3 nodes in a PD cluster. If a PD node is down for over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator takes this node offline first, and creates a new PD node. At this time, there are 4 nodes in the cluster. If the failed PD node gets back online, TiDB Operator deletes the newly created node and the number of nodes gets back to 3.
```

File: dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ pdFailoverPeriod: 5m
+ # tikv failover period default(5m)
+ tikvFailoverPeriod: 5m
+ # tidb failover period default(5m)
+ tidbFailoverPeriod: 5m
+```
+
+By default, `pdFailoverPeriod`, `tikvFailoverPeriod` and `tidbFailoverPeriod` are set to be 5 minutes, which is the waiting timeout after confirming the instance failure. After this time, TiDB Operator begins the automatic failover process.
+
+## Implementation principles
```
Comment (dcalvin) [inline dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md]: Maybe we should title it "Automatic failover policies", and for each component, just use the component name as the header.

File: dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+### PD automatic failover policy
+
+Assume that there are 3 nodes in a PD cluster, and if a PD node is down over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator makes this node go offline first, and creates a new PD node. At this time, there are 4 Pods existing in the cluster. After the failed PD node gets back to normal, TiDB Operator deletes the newly created node and the number of nodes gets back to 3.
+
+### TiKV automatic failover policy
+
+When a TiKV node fails, its status turns to `Disconnected`. After 30 minutes (configurable by modifying `pd.maxStoreDownTime` when deploying the cluster), it turns to `Down`. After waiting for 5 minutes (configurable by modifying `tikvFailoverPeriod`), TiDB Operator creates a new TiKV node if this TiKV node is still down. When the failed TiKV node gets back to normal, TiDB Operator does not automatically delete the newly created node, and you need to manually drop it and make the number of nodes the same as before. To do that, you can delete the TiKV node from the `status.tikv.failureStores` field of the `TidbCluster` object:
```
Comment (dcalvin) [inline dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md]:
```suggestion
When a TiKV node fails, its status turns to `Disconnected`. After 30 minutes (configurable by modifying `pd.maxStoreDownTime` when deploying the cluster), it turns to `Down`. After waiting for 5 minutes (configurable by modifying `tikvFailoverPeriod`), TiDB Operator creates a new TiKV node if this TiKV node is still down. If the failed TiKV node gets back online, TiDB Operator does not automatically delete the newly created node. You need to manually drop it and restore the original number of nodes. To do this, you can delete the TiKV node from the `status.tikv.failureStores` field of the `TidbCluster` object:
```

File: dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    failureStores:
+      "2":
+        podName: cluster1-tikv-1
+        storeID: "2"
+...
+```
+
+### TiDB automatic failover policy
+
+When there are 3 nodes in a TiDB cluster, the TiDB automatic failover policy is the same as the `deployment` behavior in Kubernetes. If a TiDB node is down over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator creates a new TiDB node. At this time, there are 4 Pods existing in the cluster. When the failed TiDB node gets back to normal, TiDB Operator deletes the newly created node and the number of nodes gets back to 3.
```
Comment (dcalvin) [inline dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md]:
```suggestion
The TiDB automatic failover policy works the same way as `deployment` does in Kubernetes. Assume that there are 3 nodes in a TiDB cluster. If a TiDB node is down over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator creates a new TiDB node. At this time, there are 4 nodes in the cluster. If the failed TiDB node gets back online, TiDB Operator deletes the newly created node and the number of nodes gets back to 3. 
```

**Comment by yikeke** [comment]
Comments addressed. PTAL again @dcalvin 
PTAL @weekface

File: dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
+---
+title: Automatic Failover
+summary: Learn the automatic failover policies of TiDB cluster components on Kubernetes.
+category: how-to
+---
+
+# Automatic Failover
+
+Automatic failover means that when a node in the TiDB cluster fails, TiDB Operator automatically adds a new one to ensure the high availability of the cluster. It works similarly with the `deployment` behavior in Kubernetes.
```
Comment (weekface) [inline dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md]:
```suggestion
Automatic failover means that when a node in the TiDB cluster fails, TiDB Operator automatically adds a new one to ensure the high availability of the cluster. It works similarly with the `Deployment` behavior in Kubernetes.
```

File: dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    failureStores:
+      "2":
+        podName: cluster1-tikv-1
+        storeID: "2"
+...
+```
+
+### Failover with TiDB
+
+The TiDB automatic failover policy works the same way as `deployment` does in Kubernetes. Assume that there are 3 nodes in a TiDB cluster. If a TiDB node is down for over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator creates a new TiDB node. At this time, there are 4 nodes in the cluster. When the failed TiDB node gets back online, TiDB Operator deletes the newly created node and the number of nodes gets back to 3.
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/auto-failover.md]:
```suggestion
The TiDB automatic failover policy works the same way as `Deployment` does in Kubernetes. Assume that there are 3 nodes in a TiDB cluster. If a TiDB node is down for over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator creates a new TiDB node. At this time, there are 4 nodes in the cluster. When the failed TiDB node gets back online, TiDB Operator deletes the newly created node and the number of nodes gets back to 3.
```

File: v3.0/how-to/maintain/tidb-in-kubernetes/auto-failover.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
+---
+title: Automatic Failover
+summary: Learn the automatic failover policies of TiDB cluster components on Kubernetes.
+category: how-to
+---
+
+# Automatic Failover
+
+Automatic failover means that when a node in the TiDB cluster fails, TiDB Operator automatically adds a new one to ensure the high availability of the cluster. It works similarly with the `deployment` behavior in Kubernetes.
```
Comment (yikeke) [inline v3.0/how-to/maintain/tidb-in-kubernetes/auto-failover.md]:
```suggestion
Automatic failover means that when a node in the TiDB cluster fails, TiDB Operator automatically adds a new one to ensure the high availability of the cluster. It works similarly with the `Deployment` behavior in Kubernetes.
```

File: v3.0/how-to/maintain/tidb-in-kubernetes/auto-failover.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    failureStores:
+      "2":
+        podName: cluster1-tikv-1
+        storeID: "2"
+...
+```
+
+### Failover with TiDB
+
+The TiDB automatic failover policy works the same way as `deployment` does in Kubernetes. Assume that there are 3 nodes in a TiDB cluster. If a TiDB node is down for over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator creates a new TiDB node. At this time, there are 4 nodes in the cluster. When the failed TiDB node gets back online, TiDB Operator deletes the newly created node and the number of nodes gets back to 3.
```
Comment (yikeke) [inline v3.0/how-to/maintain/tidb-in-kubernetes/auto-failover.md]:
```suggestion
The TiDB automatic failover policy works the same way as `Deployment` does in Kubernetes. Assume that there are 3 nodes in a TiDB cluster. If a TiDB node is down for over 5 minutes (configurable by modifying `tidbFailoverPeriod`), TiDB Operator creates a new TiDB node. At this time, there are 4 nodes in the cluster. When the failed TiDB node gets back online, TiDB Operator deletes the newly created node and the number of nodes gets back to 3.
```

**Pull request merged**